packages/flutter/lib/src/widgets/animated_child_switcher.dart

```diff
@@ -73,10 +220,49 @@ class _AnimatedChildSwitcherState extends State<AnimatedChildSwitcher> with Tick
   @override
   void initState() {
     super.initState();
-    addEntry(false);
+    addEntry(animate: false);
   }
 
-  void addEntry(bool animate) {
+  Widget _generateTransition(Animation<double> animation) {
+    return new KeyedSubtree(
+      key: new UniqueKey(),
+      child: widget.transitionBuilder(widget.child, animation),
+    );
+  }
+
+  _AnimatedChildSwitcherChildEntry _newEntry({
+    @required AnimationController controller,
+    @required Animation<double> animation,
+  }) {
+    final _AnimatedChildSwitcherChildEntry entry = new _AnimatedChildSwitcherChildEntry(
+      widgetChild: widget.child,
+      transition: _generateTransition(animation),
+      animation: animation,
+      controller: controller,
+    );
+    animation.addStatusListener((AnimationStatus status) {
+      if (status == AnimationStatus.dismissed) {
+        assert(_children.contains(entry));
+        setState(() {
+          _children.remove(entry);
+        });
+        controller.dispose();
+      }
+    });
+    return entry;
+  }
+
+  void addEntry({@required bool animate}) {
```

Comment:
If _newEntry is private, shouldn't addEntry be _addEntry?

Reply:
Done.